lib/src/geo/crs.dart

```diff
@@ -13,24 +13,34 @@ import 'package:proj4dart/proj4dart.dart' as proj4;
 /// points of objects of different dimensions. In our case 3D and 2D objects.
 @immutable
 abstract class Crs {
```

Comment:
I don't know. Do you want to prevent folks from implementing their own coordinate systems? :shrug: Would definitely break any implementations that already exist.

Reply:
Would this prevent them from implementing their own? Afaik `base` doesn't imply `final`, it just forces extension (instead of implementation) only, which seems to be what you want, as you are using `@nonVirtual`?

Reply:
You're correct that base and final are different. My statement is based on the documentation "A base class disallows implementation outside of its own library." https://dart.dev/language/class-modifiers

Reply:
But `implements`ation doesn't make sense if any fields are marked `nonVirtual`. So it would seem to me as though `base` is just reinforcing what you already want.

Reply:
Sorry, maybe I'm terribly confused or we're talking past each other.

What I understand atm: I made a few abstract methods members and marked them as nonVirtual because they no longer need to be implemented. Yet, there are still abstract methods left that need implementing. There are several CRS implementations in this file. AFAIU, adding the "base" class-modifier won't do anything other than disallow implementations of CRS outside this module.

If that's what we want, I'll add the modifier. I don't understand how this relates to some members being marked @nonVirtual.

Reply:
Whoops, totally another mistake from me. I thought `@nonVirtual` will cause a lint error for classes that `implements` rather than `extends`, but nope!